packages/NodeCollector/BinaryOpConditionsCollector.php

```diff
@@ -19,9 +19,9 @@ final class BinaryOpConditionsCollector
      * the implicit parenthesization (`[a, b ?? c]`) might not match the expectations.
      *
      * @param class-string<BinaryOp> $binaryOpClass
-     * @return Expr[]
+     * @return array<int, Expr>
      */
-    public function findConditions(BinaryOp $binaryOp, string $binaryOpClass): array
+    public function findConditions(Expr $binaryOp, string $binaryOpClass): array
```

Comment:
I think `BinaryOp` should be kept to ensure the usage is on `BinaryOp`

Reply:
See the [last commit](https://github.com/rectorphp/rector-src/commit/ad8194ac1c6e5b2b979f3b5d1f9df8a5bbafc989), it makes for much nicer definitions that way.

Reply:
Especially with https://github.com/rectorphp/rector-src/commit/b76243b41682edd072da3ca149461283c779415b.

Reply:
I added a test case and comment making it clearer.